tyr-runner/src/main/java/org/jboss/tyr/api/GitHubAPI.java

```diff
@@ -119,14 +120,22 @@ static JsonReader getJSONReader(URI uri) {
         }
     }
 
-    private static URI getCommitsUri(JsonObject prPayload) {
-        String url = prPayload.getJsonObject(Utils.PULL_REQUEST).getString(Utils.COMMITS_URL);
-        return URI.create(url);
+    private static URI getCommitsUri(JsonObject prPayload) throws InvalidPayloadException {
+        try {
+            String url = prPayload.getJsonObject(Utils.PULL_REQUEST).getString(Utils.COMMITS_URL);
+            return URI.create(url);
+        } catch (NullPointerException e) {
+            throw new InvalidPayloadException("Invalid payload, can't retrieve URL. Message was: " + e.getMessage());
```

Comment:
wouldn't it be better to add NPE as the cause of IPE?

Reply:
sure...something like that: "Failed with NullPointerException, can't retrieve URL. Message was: .." ?